workflow-core/src/main/java/com/squareup/workflow/Sink.kt

```diff
@@ -47,3 +52,61 @@ fun <T1, T2> Sink<T1>.contraMap(transform: (T2) -> T1): Sink<T2> {
     }
   }
 }
+
+/**
+ * Collects from a [Flow] by converting each item into a [WorkflowAction] and then sending them
+ * to the [actionSink]. This operator propagates backpressure from the workflow runtime, so if there
+ * is a lot of contention on the workflow runtime the flow will be suspended while the action is
+ * queued.
+ *
+ * This method is intended to be used from [RenderContext.runningSideEffect].
+ *
+ * Example:
+ * ```
+ * context.runningSideEffect("collector") {
+ *   myFlow.collectToSink(context.actionSink) { value ->
+ *     action { setOutput(value) }
+ *   }
+ * }
+ * ```
+ */
+@ExperimentalWorkflow
+suspend fun <T, StateT, OutputT : Any> Flow<T>.collectToSink(
+  actionSink: Sink<WorkflowAction<StateT, OutputT>>,
+  handler: (T) -> WorkflowAction<StateT, OutputT>
+) {
+  collect {
+    // Don't process the emission until the last emission has had its action executed by the
+    // workflow runtime.
+    actionSink.sendAndAwaitApplication(handler(it))
+  }
+}
+
+/**
+ * Sends [action] to this [Sink] and suspends until after [action]'s [WorkflowAction.apply] method
+ * has been invoked. Since a [Sink] may be backed by an unbounded buffer, this method can be used
+ * to apply backpressure to the caller when there are a lot events being sent to the workflow
+ * runtime.
+ *
+ * If this coroutine is cancelled before the action gets applied, the action will not be applied.
+ *
+ * This method is intended to be used from [RenderContext.runningSideEffect].
```

Comment:
sample use?

Reply:
Not sure what value a sample would add: there's really only one possible way to use this method.